frontend/fingertips-frontend/components/atoms/AuthButton/index.tsx

```diff
@@ -11,16 +13,38 @@ export function AuthButton({ session }: Readonly<AuthButtonProps>) {
 }
 
 function SignInButton() {
+  const { setIsLoading } = useLoadingState();
+  useEffect(() => {
+    setIsLoading(false);
+  });
+
   return (
-    <StyledAuthButton data-testid="sign-in-button" onClick={signInHandler}>
+    <StyledAuthButton
+      data-testid="sign-in-button"
+      onClick={() => {
+        setIsLoading(true);
+        signInHandler();
+      }}
+    >
       Sign in
     </StyledAuthButton>
   );
 }
 
 function SignOutButton() {
+  const { setIsLoading } = useLoadingState();
+  useEffect(() => {
+    setIsLoading(false);
+  });
```

Comment:
does the rendering of this button guarantee that all loading has finished?
```suggestion
  useEffect(() => {
    setIsLoading(false);
  }, []); // missing dependencies
```